### PR TITLE
[15.0][FIX] hr_employee_document: Remove record rules from hr.employee to avoid side effects..

### PR DESCRIPTION
**Pull request by victoralmau**
Remove record rules from hr.employee to avoid side effects.

**Example of use case**:
- An employee X is created and a Manager Y (`parent_id`) is defined.
- User X creates an expense report (`hr.expense.sheet`).
- The "_Submit to manager_" button is clicked.
- A permissions error is displayed when accessing employee Y (`hr.employee`) because the record rules that this module had added do not allow access to that record.

**Solution**:
This change overrides some things to allow the user's employee attachments to be displayed.

Please @pedrobaeza can you review it?

@Tecnativa TT44536

### CODE REVIEW

**Comment by pedrobaeza** [comment]
You have to put the steps to reproduce the problem in the commit message.

Apart from that, why that ACLs was there? (isn't they record rules, not ACL?) Seeing a bit the code, it seems you are solving it acting at code level. Please explain all of this in the commit message, docstring, etc.

**Comment by victoralmau** [comment]
Sorry, changed the commit message and explained a use case.

File: hr_employee_document/models/ir_attachment.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class IrAttachment(models.Model):
+    _inherit = "ir.attachment"
+
+    @api.model
+    def _search(
```
Comment (pedrobaeza) [inline hr_employee_document/models/ir_attachment.py]: Security shouldn't be handled in `_search` method. This is a security hole, as you can browse directly some random records.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by victoralmau** [comment]
I have changed the approach to try to solve the problem in another way, although there is still some detail to be solved, what do you think about this way?

File: hr_employee_document/models/hr_employee.py [hunk 2]
```diff
@@ -22,15 +23,33 @@ def _compute_document_count(self):
...
+        """Return access to the hr.employee model if we pass a specific context,
+        is a trick to list the attachments related to an employee."""
+        if (
+            not self.env.is_superuser()
+            and not self.env.user.has_group("hr.group_hr_user")
+            and operation == "read"
+            and self._name == "hr.employee"
+        ):
+            if (
+                self.env.context.get("search_attachments_from_hr_employee")
```
Comment (pedrobaeza) [inline hr_employee_document/models/hr_employee.py:37]: Shouldn't this be limited to the specific employee?

File: hr_employee_document/models/ir_rule.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            "mode",
+            "tuple(self._compute_domain_context_values())",
+        ),
+    )
+    def _compute_domain(self, model_name, mode="read"):
+        """We need to add for security purposes an extra domain in the hr.employee
+        model to restrict only the user's employees when search employee attachments."""
+        res = super()._compute_domain(model_name, mode=mode)
+        user = self.env.user
+        if model_name == "hr.employee" and not self.env.su:
```
Comment (pedrobaeza) [inline hr_employee_document/models/ir_rule.py]: Shouldn't this also exclude HR managers?

Comment (victoralmau) [reply hr_employee_document/models/ir_rule.py]: Now the extra domain is not added to the rule if the user has the Employees > Officer permission (this is the same behavior as the module currently has).

Comment (pedrobaeza) [reply hr_employee_document/models/ir_rule.py]: But our use case wants to exclude Officers. We want this only for HR managers, not officers.

**Review by pedrobaeza — APPROVED** [review]

**Review by chienandalu — APPROVED** [review]
LGTM :+1:

**Pull request merged**